CIPs/0011.md

```diff
@@ -0,0 +1,112 @@
+# CIP 11: Off-chain claims and attestations
+
+- Date: 2020-05-20
+- Author: @nambrot
+- Status: WIP
+
+## Overview
+
+Celo's lightweight identity protocol uses the `Attestations` smart contract to store requests of attestations to a random set of validators and their completions. Through consensus on the randomness of the selection, the sybil resistance provided by the proof of stake acts as a "commoditizing" force and ultimately tries to convey an attestation by the protocol to a certain extent.
+
+Fundamentally, attestations generally however are contextual to the attestor and do not require consensus (which is expensive). Think about the Celo foundation attesting stake-off winners. Users on the network do not need to agree on the order of these attestations, all they need is the cryptographic attestation (signature) by the foundation's key. Other direct attestations can include security-audits by audit firms, attestations by validators to convey helpful peers or attestations to releases. Attestations can also form the backbone of name services, modeled as attestations by the registrar.
+
+Another form of an attestation is produce an account's signature and then have the attestor (such as a social media network) host it in a public way. That effectively acts as an attestation by attestor in a particular namespace (such as a profile). On Celo, an account can then make a claim to that attestation and any user can verify that attestation by checking the attestor without needing protocol-level consensus. Celo's `metadata` feature currently allows accounts to make claims over Keybase accounts or domain names. Metadata was originally intended for more narrow use cases.
+
+This CIP tries to standardize how accounts can make claims and attestations by migrating from it's original location in `metadata` to generalized off-chain storage as proposed in CIP8.
+
+## Goals
+
+- Provide a flexible standard for attestations
+- Provide a flexible standard for claims that can be verified as attestations
+
+## Proposed Solution
+
+On top of CIP8, Attestations for an account should be made under `/accounts/${otherAccountAddress}/attestations/${attestationType}`.
+
+```json
+{
+  "type": "${attestationType}",
+  "attestor": "${accountAddress}",
+  "attestee": "${otherAccountAddress}",
+}
+```
+
+We define the following attestation types:
+
+```json
+{
+  "type": "attestation_TGCSO_attestation_maven"
+}
+```
+
+```json
+{
+  "type": "attestation_TGCSO_founder_validator",
+  "score": ""
+}
+```
+
+```json
+{
+  "type": "attestation_TGCSO_master_validator",
```

Comment:
also: `attestation_mainnet_first_proposer_validator`, `attestation_mainnet_first_transaction_validator`

Reply:
Since that data already lives on-chain, it feels superflous here? Also I'm not sure that one getting lucky as the proposer should really be a worthwhile attestation?